File: Java/DataStructures/LinkedList.java
public class LinkedList {
    
    public class Node{
    
    int data;
    Node next;
    
    public Node(int data){
        this.data = data;
        this.next = null;
    }

    public int getData(){
        return this.data;
    }

    public void setData(int data){
        this.data = data;
    }

    public Node getNext(){
        return this.next;
    }

    public void setNext(Node next){
        this.next = next;
    }
}
    Node head;
    
    void pushFront(int data){
        Node node = new Node(data);
        node.setNext(head);
        head = node;
    }
    
    int topFront(){
        
        return head.getData();
    }
    
    void popFront(){
        head = head.getNext();
    }
    
    void pushBack(int data){
        Node current = head;
        Node node = new Node(data);
        if(head == null){
            head = node;
        }else{
            while(current.getNext() != null){
                current = current.getNext();
            }
            current.setNext(node);
        }
    }   
    
    int topBack(){
        Node current = head;
        if(head == null){
            return 0;
        }
        while(current.getNext() != null){
            current = current.getNext();
        }
        return current.getData();
    }
    
    void popBack(){
        Node current = head;
        if(head == null){
            System.out.println("List Empty");
        }else{
            Node prev = current;
            while(current.getNext() != null){
                prev = current;
                current = current.getNext();
            }
            
            prev.setNext(null);
        }
    }
    
    boolean find(int data){
        Node current = head;
        if(head == null){
            return false;
        }else{
            while(current != null){
                
                if(current.getData() == data){
                    return true;
                }else{
                    current = current.getNext();   
                }
            }
        }  
        return false;
    }
    
    boolean Empty(){
        if(head == null){
            return true;
        }
        return false;
    }
    
    void printList(){
        Node current = head;
        if(head == null){
            System.out.println("List Empty");
        }else{
            while(current != null){  
                System.out.println(current.getData());
                current = current.getNext();
            }
        }
    }
}

File: C++/DataStructures/LinkedList.cpp
#include <iostream>

class Node{

    private: 
    
    int data;
    Node *next;
    
    public:
    
    Node(int data){
        this->data = data;
        this->next = nullptr;
    }

    int getData(){
        return this->data;
    }

    void setData(int data){
        this->data = data;
    }

    Node *getNext(){
        return this->next;
    }

    void setNext(Node *next){
        this->next = next;
    }
};  

class LinkedList{

    Node *head = nullptr;

    public:
    LinkedList(){
        this->head = nullptr;
    }

    void pushBack(int data){
        Node *newNode = new Node(data);
        Node *ptr = this->head;
        if(this->head == nullptr){
            this->head = newNode;
        }else{
            while(ptr->getNext() != nullptr){
                ptr = ptr->getNext();
            }
            ptr->setNext(newNode);
        }
    }

    void popBack(){
        Node* ptr = this->head;
        Node* prev = nullptr;
        if(ptr == nullptr){
            std:: cout << "List Empty" <<std::endl;
        }else{
            while(ptr->getNext() != nullptr){
                prev = ptr;
                ptr = ptr->getNext();
            }
            prev->setNext(nullptr);
        }
    }

    void pushFront(int data){
        Node *newNode = new Node(data);
        newNode->setNext(this->head);
        this->head = newNode;  
    }

    void printList(){
        Node *ptr = this->head;
        if(ptr == nullptr){
            std:: cout << "List Empty" <<std::endl;
        }else{
            while(ptr != nullptr){
                std:: cout << ptr->getData() << std::endl;
                ptr = ptr->getNext();
            }
        }
    }
};


int main() {

    LinkedList myList;

    myList.pushBack(1);
    myList.pushBack(3);
    myList.pushBack(2);
    std::cout << "pushBack" <<std::endl; 
    myList.printList();

    myList.popBack();
    std::cout << "popBack" <<std::endl; 
    myList.printList();

    myList.pushFront(2);
    std::cout << "pushFront" <<std::endl; 
    myList.printList();

    std::cout << "hello word" <<std::endl; 
    return 0;
}
